### PR TITLE
Api end point

### PR DESCRIPTION
**Pull request by Smart1-hub**
Create an API endpoint to list all posts for a user.
    Create an API endpoint to list all comments for a user's post.
    Create an API endpoint to add a comment to a post. 
    Add authentication to your API endpoints.
    API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -66,7 +69,13 @@ end
...
+  gem 'capybara'
+  gem 'selenium-webdriver'
+  gem 'webdrivers'
 end
+
+gem 'bullet', group: 'development'
+
+gem 'devise'
+
+gem 'cancancan'
```
Comment (rotimiazeez) [inline Gemfile:81]: - When I try setting up the database at my end, I'm met with the below error.
Screenshot from 2022-09-09 10-29-20
Please be sure to add `Rswag api` and `Rswag ui ` to your gem file to fix this 👍🏽

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
-
+  devise_for :users
   # Defines the root path route ("/")
-  # root "articles#index"
+  root 'users#index'
+
+  resources :users, only: %i[index show] do
+    resources :posts, only: %i[index new create show destroy] do
+    end
+  end
```
Comment (rotimiazeez) [inline config/routes.rb:12]: - On starting your app I get the below error. I believe it has to do with your asset file, kindly fix it.
Screenshot from 2022-09-09 10-34-53

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Smart1-hub 👏🏽, 
## Good job so far! 👍🏽 
### Summary ♨️ 
- No Linter error ✔️ 
- Good and descriptive PR 💯 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Smart1-hub — COMMENTED** [review]
Update dependencies and fix connection error as requested by the reviewer

**Review by brytebee — APPROVED** [review]
Hi Team👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great implementation of suggestion by previous reviewer.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — COMMENTED** [review]
Hi @Smart1-hub, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**